packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -65,10 +67,15 @@ export default function useNestedSettingsUpdate(
 	// Memoize as inner blocks implementors often pass a new array on every
 	// render.
 	const _allowedBlocks = useMemo( () => allowedBlocks, allowedBlocks );
+	const _disallowedBlocks = useMemo(
+		() => __unstableDisallowedBlocks,
+		__unstableDisallowedBlocks
+	);
```

Comment:
Both these two memoize calls are very weird 🤔

Reply:
Yes. I updated them.

Reply:
Oh sorry I confused with my comment, I think the previous code was intended, and your update just make the useMemo useless. 

it's just sad that we need this code :)

Reply:
Reverted again :). I don't really quite understand the reason behind this 🤔

Reply:
I think it's to avoid rendering when people write  `allowedBlocks={ [ "something", "something" }` in their components. Since the prop there is using an inline array (new instance on each render), if there's a sub component that consumes this prop but is memoized (using React.memo or pure), it will get re-rendered on each render anyway